src/world/mod.rs

```diff
@@ -193,8 +194,8 @@ impl World {
         let archetype_idx =
             cached_archetype_idx(&mut self.keys, &mut self.ids, &mut self.archetypes, &bundle);
 
-        self.epoch += 1;
-        let idx = self.archetypes[archetype_idx as usize].spawn(entity, bundle, self.epoch);
+        let epoch = self.epoch.fetch_add(1, Ordering::Relaxed).wrapping_add(1);
```

Comment:
You don't need to perform increments atomically in methods that takes `&mut self`.
Use `AtomicU64::get_mut` to take a mutable reference to the underlying integer and mutate as usual.